internal/controller/instaslice_daemonset.go

```diff
@@ -109,37 +115,25 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
 	}
 
 	emulatorMode := os.Getenv("EMULATOR_MODE")
-	log.FromContext(ctx).Info(" daemonset simulator mode ", "enabled", emulatorMode)
-
+	log.FromContext(ctx).Info("daemonset simulator mode ", "enabled", emulatorMode)
 	for _, allocations := range instaslice.Spec.Allocations {
 		//TODO: we make assumption that resources would always exists to delete
 		// if user deletes abruptly, cm, instaslice resource, ci and gi may not exists
 		// handle such scenario's.
 		// delete first before creating new slice
 		if allocations.Allocationstatus == inferencev1alpha1.AllocationStatusDeleting && allocations.Nodename == nodeName {
-			log.FromContext(ctx).Info("Performing cleanup ", "pod", allocations.PodName)
-			extendedResourceName := AppendToInstaSlicePrefix(allocations.Resourceidentifier)
+			log.FromContext(ctx).Info("performing cleanup ", "pod", allocations.PodName)
 			if errDeletingCm := r.deleteConfigMap(ctx, allocations.Resourceidentifier, allocations.Namespace); errDeletingCm != nil {
-				log.FromContext(ctx).Error(errDeletingCm, "error deleting configmap for ", "pod", allocations.PodName)
-				return ctrl.Result{Requeue: true}, nil
-			}
-			if errDeletingInstaSliceResource := r.cleanUpInstaSliceResource(ctx, extendedResourceName); errDeletingInstaSliceResource != nil {
-				log.FromContext(ctx).Error(errDeletingInstaSliceResource, "Error deleting InstaSlice resource object")
-				return ctrl.Result{Requeue: true}, nil
-			}
-
-			nodeName := os.Getenv("NODE_NAME")
-			if errUpdatingNodeCapacity := r.updateNodeCapacity(ctx, nodeName, allocations, emulatorMode); errUpdatingNodeCapacity != nil {
 				return ctrl.Result{Requeue: true}, nil
 			}
-			var deletePrepared string
 
 			if emulatorMode == emulatorModeFalse {
-				deletePrepared = r.cleanUpCiAndGi(ctx, allocations.PodUUID, instaslice)
-				log.FromContext(ctx).Info("Done deleting ci and gi for ", "pod", allocations.PodName)
+				errCleaningCiAndGi := r.cleanUpCiAndGi(ctx, allocations.PodUUID, instaslice)
+				if errCleaningCiAndGi != nil {
+					return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
```

Comment:
error is not returned, could be just `err`

Reply:
need to wait before we retry, this is a transient error where the NVML init sometimes in flaky. I will change the error variable

Reply:
Sometimes NVML shutdown takes time, hence we delay the requeueing and do not return the error. this is transient. let me add a logger for it.